### PR TITLE
Update tuya.ts

### PR DESCRIPTION
**Pull request by Osrx**
Fix issues in commit from @alexelis
#7604

Added support for ElectSmart PRO-900Z electric floor thermostat.

### CODE REVIEW

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -10600,6 +10600,113 @@ const definitions: Definition[] = [
...
+                })],
+                [2, 'preset', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'AUTO': tuya.enum(0), 'MANUAL': tuya.enum(1)};
+                })],
+                [16, 'current_heating_setpoint', tuya.valueConverter.divideBy10],
+                [19, 'max_temperature', tuya.valueConverter.divideBy10],
+                [24, 'local_temperature', tuya.valueConverter.divideBy10],
+                [26, 'min_temperature', tuya.valueConverter.divideBy10],
+                [27, 'local_temperature_calibration', tuya.valueConverter.raw],
+                [28, 'factory_reset', tuya.valueConverterBasic.lookup((_, device) => {
```
Comment (Koenkk) [inline src/devices/tuya.ts:10680]:
```js
[28, 'factory_reset', tuya.valueConverterBasic.lookup({'factory_reset': true}],
```

Please use this, also for the other `tuya.valueConverterBasic.lookup`

File: src/devices/tuya.ts [hunk 2]
```diff
@@ -10600,6 +10600,113 @@ const definitions: Definition[] = [
...
+            e.enum('factory_reset', ea.STATE_SET, ['factory reset'])
+                .withLabel('Factory reset').withDescription('Reset all settings to factory ones'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'system_mode', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'OFF': false, 'HEAT': true};
+                })],
+                [2, 'preset', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'AUTO': tuya.enum(0), 'MANUAL': tuya.enum(1)};
```
Comment (Koenkk) [inline src/devices/tuya.ts:10673]:
```suggestion
                    return {'auto': tuya.enum(0), 'manual': tuya.enum(1)};
```

File: src/devices/tuya.ts [hunk 3]
```diff
@@ -10600,6 +10600,113 @@ const definitions: Definition[] = [
...
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_sunday', ea.STATE_SET).withLabel('Schedule for sunday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.enum('factory_reset', ea.STATE_SET, ['factory reset'])
+                .withLabel('Factory reset').withDescription('Reset all settings to factory ones'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'system_mode', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'OFF': false, 'HEAT': true};
```
Comment (Koenkk) [inline src/devices/tuya.ts:10670]:
```suggestion
                    return {'off': false, 'heat': true};
```

File: src/devices/tuya.ts (hunk 1 above)
Comment (Osrx) [reply src/devices/tuya.ts]: Thank you for this. I'm not a programmer, but I'll try.
Fixed.

File: src/devices/tuya.ts [hunk 4]
```diff
@@ -10707,6 +10707,101 @@ const definitions: Definition[] = [
...
+        }],
+        model: 'PRO-900Z',
+        vendor: 'ElectSmart',
+        description: 'Thermostat for electric floor',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        onEvent: tuya.onEventSetTime,
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            e.enum('child_lock', ea.STATE_SET, ['LOCK', 'UNLOCK']).withLabel('Child lock'),
```
Comment (Koenkk) [inline src/devices/tuya.ts:10723]:
```suggestion
            e.binary('child_lock', ea.STATE_SET, 'ON', 'OFF').withLabel('Child lock'),
```

File: src/devices/tuya.ts [hunk 5]
```diff
@@ -10707,6 +10707,101 @@ const definitions: Definition[] = [
...
+            e.enum('eco_mode', ea.STATE_SET, ['OFF', 'ON']).withLabel('ECO mode').withDescription('Default: Off'),
```
Comment (Koenkk) [inline src/devices/tuya.ts:10724]: Also use `e.binary()` here.

File: src/devices/tuya.ts [hunk 6]
```diff
@@ -10707,6 +10707,101 @@ const definitions: Definition[] = [
...
+            e.numeric('eco_temperature', ea.STATE_SET).withValueMin(5).withValueMax(30).withValueStep(1).withUnit('°C')
+                .withDescription('Max temperature in ECO mode. Default: 20'),
+            e.text('valve_state', ea.STATE).withLabel('Heating in process'),
```
Comment (Koenkk) [inline src/devices/tuya.ts:10727]: Since it's a binary (`[36, 'valve_state', tuya.valueConverter.trueFalseInvert],`), this should also be `e.binary()`

**Comment by Koenkk** [comment]
Could you also submit a picture for the docs? This can be done by:
1. Create a fork by clicking [here](https://github.com/Koenkk/zigbee2mqtt.io/fork)
2. Go to the `public/images/devices` directory, *Add file* -> *Upload files*
3. Upload the files and press *Commit changes*
4. Press *Contribute* -> *Open pull request* -> update title/description -> *Create pull request*

Make sure that:
- The filename is `PRO-900Z.png`
- The size is 512x512
- The background is transparent (use e.g. [Adobe remove background](https://new.express.adobe.com/tools/remove-background))

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**